### PR TITLE
[AP-3085] Use local github action to delete UAT release

### PR DESCRIPTION
**Pull request by jsugarman**
## What
Delete UAT deployments on branch closure or merge

[Link to story](https://dsdmoj.atlassian.net/browse/AP-3085)

L&D

We automatically create UAT deployments on opening a PR. They are automatically 
deleted by CircleCI steps on merge but not on branch closure.

## Checklist
- [X] on merge PR - to be tested
- [X] on close PR.
- [X] add readme
- [ ] extract to own repo?!

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: .github/actions/delete-uat-release/action.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+name: "Delete UAT deployment"
+description: 'Deletes a UAT deployment with a name mathcing the merged/closed branch'
```
Comment (colinbruce) [inline .github/actions/delete-uat-release/action.yml:2]: typo : matching

File: .github/workflows/delete-uat-release.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: Delete UAT release
+
+on:
+  pull_request:
+    types:
+      - closed
+
+jobs:
+  merge_job:
+    if: github.event.pull_request.merged == true
```
[the remark this answers is not in the document]
Comment (colinbruce) [reply .github/workflows/delete-uat-release.yml:10]: Agree :+1:

**Review by colinbruce — APPROVED** [review]
Looks okay :+1: 
One nitpicky typo and I agree with maybe combining merge & close

**Comment by jsugarman** [comment]
Thanks colin, was also thinking to extract to its own repo for apply specific use. currently have it in my own account, being used in this [PR  here for example](https://github.com/ministryofjustice/laa-apply-for-legal-aid/pull/3886/files#diff-a803634c13b1fcc25484c2444933d811ec5c686c279167e0de0c4e1372c6a177R16)

Could then be used centrally across all apply "services" with a little tweaking relating to UAT release name. 

I found a repo you had created for shared github actions but cannot locate now?!

**Pull request merged**